File: backend/app/models/annotation.py
import datetime

from sqlalchemy import (
    Column, Integer, String, Text, DateTime,
    ForeignKey, func,
)

from app.db.base import Base


class Annotation(Base):
    __tablename__ = "annotations"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    document_id = Column(Integer, ForeignKey("documents.id"), nullable=False, index=True)
    page_number = Column(Integer, default=1)
    annotation_type = Column(String(16), default="highlight")  # highlight | underline | comment
    color = Column(String(32), default="#FFD700")
    selected_text = Column(Text, default="")
    note = Column(Text, default="")
    position_data = Column(Text, default="{}")  # JSON string
    range_data = Column(Text, default="{}")  # JSON string
    tags = Column(String(256), default="")
    deleted_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, server_default=func.now())

File: backend-go/internal/models/annotation.go
package models

import "time"

type Annotation struct {
	ID             uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID         uint      `gorm:"index;not null" json:"user_id"`
	DocumentID     uint      `gorm:"index;not null" json:"document_id"`
	PageNumber     int       `gorm:"default:1" json:"page"`
	AnnotationType string    `gorm:"size:16;default:'highlight'" json:"type"`
	Color          string    `gorm:"size:32;default:'#FFD700'" json:"color"`
	SelectedText   string    `gorm:"type:text" json:"selected_text"`
	Note           string    `gorm:"type:text" json:"note"`
	PositionData   string    `gorm:"type:text" json:"position_data"`
	DeletedAt      *time.Time `gorm:"index" json:"-"`
	CreatedAt      time.Time  `json:"created_at"`
}
